src/compiler/transformers/module/module.ts

```diff
@@ -88,11 +88,14 @@ namespace ts {
             addRange(statements, endLexicalEnvironment());
             addExportEqualsIfNeeded(statements, /*emitAsReturn*/ false);
 
+            if (!currentModuleInfo.exportEquals) {
+                append(statements, createUnderscoreUnderscoreESModule());

```

Comment:
Not really familiar with TS codebase to know if this is easily fixable, but the `__esModule` marker should be created _as soon as possible_. It needs to be visible from circular imports.